### ISSUE TITLE
Image.network widget cannot rending the  gzip encoded image

### ISSUE THREAD

**Issue by panio123** [flutter/flutter#39573]
All HttpClient requests set the header by defalt 

```
Accept-Encoding:gzip
```

so the Image.network received a gzip encoded image is possible,but it cannot rending.
I tried reset **Accept-Encoding** value to **identity** was defeated

```
Image.network(
            url
            headers: {"accept-encoding": "identity"},
          )
```

becase that:

```
in ...lib/src/painting/_network_image_io.dart

 headers?.forEach((String name, String value) {
        request.headers.add(name, value);
      });
```

finally,you only reset the **Accept-Encoding** value to **gzip,identity**

If use **headers.set** instead of **headers.add** everything is OK.

**Comment by BondarenkoStas** [comment]
@panio123 can you please provide some reproducible code example? Is it correct that the problem is only in headers.add?

**Comment by panio123** [comment]
> @panio123 can you please provide some reproducible code example? Is it correct that the problem is only in headers.add?

Yes,the ``headers.add`` method only push a new value,but in this case I want to delete the default value ``gzip ``

**Comment by markusaksli-nc** [comment]
https://github.com/flutter/flutter/blob/e8c942104bb4b414b67e303a54b00e8fc45c67a6/packages/flutter/lib/src/painting/_network_image_io.dart#L91-L93
Seems like this still hasn't been changed on the latest master.
